packages/vitest-pool-workers/src/pool/index.ts

```diff
@@ -0,0 +1,815 @@
+import assert from "node:assert";
+import crypto from "node:crypto";
+import events from "node:events";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import util from "node:util";
+import { createBirpc } from "birpc";
+import * as devalue from "devalue";
+import {
+	Log,
+	LogLevel,
+	Miniflare,
+	structuredSerializableReducers,
+	structuredSerializableRevivers,
+	WebSocket,
+	kCurrentWorker,
+	kUnsafeEphemeralUniqueKey,
+} from "miniflare";
+import { createMethodsRPC } from "vitest/node";
+import { OPTIONS_PATH, parseProjectOptions } from "./config";
+import {
+	getProjectPath,
+	getRelativeProjectPath,
+	isFileNotFoundError,
+	WORKER_NAME_PREFIX,
+} from "./helpers";
+import {
+	ABORT_ALL_WORKER,
+	handleLoopbackRequest,
+	scheduleStorageReset,
+	waitForStorageReset,
+} from "./loopback";
+import {
+	ensurePosixLikePath,
+	handleModuleFallbackRequest,
+} from "./module-fallback";
+import type { WorkersPoolOptions, SourcelessWorkerOptions } from "./config";
+import type { CloseEvent, MiniflareOptions, WorkerOptions } from "miniflare";
+import type { MessagePort } from "node:worker_threads";
+import type { Readable } from "node:stream";
+import type {
+	ResolvedConfig,
+	RunnerRPC,
+	RuntimeRPC,
+	WorkerContext,
+} from "vitest";
+import type { ProcessPool, Vitest, WorkspaceProject } from "vitest/node";
+
+function structuredSerializableStringify(value: unknown): string {
+	return devalue.stringify(value, structuredSerializableReducers);
+}
+function structuredSerializableParse(value: string): unknown {
+	return devalue.parse(value, structuredSerializableRevivers);
+}
+
+// Log for verbose debug messages (e.g. RPC messages)
+let debuglog: util.DebugLoggerFunction = util.debuglog(
+	"vitest-pool-workers:index",
+	(fn) => (debuglog = fn)
+);
+// Log for informational pool messages
+const log = new Log(LogLevel.VERBOSE, { prefix: "vpw" });
+// Log for Miniflare instances, used for user code warnings/errors
+const mfLog = new Log(LogLevel.WARN);
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const DIST_PATH = path.resolve(__dirname, "..");
+const POOL_WORKER_PATH = path.join(DIST_PATH, "worker", "index.mjs");
+
+const symbolizerWarning =
+	"warning: Not symbolizing stack traces because $LLVM_SYMBOLIZER is not set.";
+const ignoreMessages = [
+	// Not user actionable
+	// TODO(someday): this is normal operation and really shouldn't error
+	"disconnected: operation canceled",
+	"disconnected: worker_do_not_log; Request failed due to internal error",
+	"disconnected: WebSocket was aborted",
+];
+function trimSymbolizerWarning(chunk: string): string {
+	return chunk.includes(symbolizerWarning)
+		? chunk.substring(chunk.indexOf("\n") + 1)
+		: chunk;
+}
+function handleRuntimeStdio(stdout: Readable, stderr: Readable): void {
```

Comment:
Is this drastically different from https://github.com/cloudflare/workers-sdk/blob/ce23ed71b9b8c6702a19db10df6adb1ae54b0dd5/packages/wrangler/src/dev/miniflare.ts#L460? Can we reuse some of the logic across both surfaces?